src/Analytics/Adapter/ActiveCampaign.php

```diff
@@ -50,17 +63,300 @@ public function getName(): string
         return 'ActiveCampaign';
     }
 
+
+    /**
+     * Checks if a contact exists and returns a bool if it does.
+     * 
+     * @param string $email
+     * @return bool|int
+     */
+    public function contactExists(string $email): bool|int
+    {
+        try {
+            $result = $this->call('GET', '/api/3/contacts', [], [
+                'email' => $email
+            ]);
+
+            $result = json_decode($result, true);
+
+            if ($result && $result['meta']['total'] > 0) {
+                return $result['contacts'][0]['id'];
+            } else {
+                return false;
+            }
+        } catch (\Exception $e) {
+            Console::error($e->getMessage());
```

Comment:
This could be scary, as we are losing much info about exceptions, like a stack trace. Could we log the whole error instead?  I can see this in multiple places.

Reply:
I've improved the error handler by porting over the appwrite error messages, let me know what you think

Reply:
Now you got me thinking if we can do it better 🤔 This is a lot of copy&paste